test/test_utilities.h

```diff
@@ -51,6 +51,21 @@ void AssignInputValuesFromVector(Tensor& A, vector<T> x) {
   }
 }
 
+template <typename T>
+// Randomly generate data with specified type to populate the Tensor
+void AssignInputValuesRandom(Tensor& A, T min, T max) {
+  auto A_flat = A.flat<T>();
+  auto A_flat_data = A_flat.data();
+  srand(static_cast<unsigned>(time(0)));
+  for (int i = 0; i < A_flat.size(); i++) {
+    T value =
+        // randomly generate a number between 0 and (max-min)
+        static_cast<T>(rand()) / static_cast<T>(RAND_MAX / (max - min));
```

Comment:
For int , the current function generates a value between [min, max-1] and not [min, max]. Is that what we want? 

If we want [min, max] range, we would want to do change to below. This will be more intuitive. 

```
static_cast<T>(rand()) / static_cast<T>(RAND_MAX / (max+1 - min));
```

It would be helpful, if we can add a comment describing the behavior of the function, so that there is no confusion for the user. 

For e.g. in the SparseCrossEntropyTest(), we need to provide labels. So the tensor should be assigned an int between [0, num_of_classes -1] . 

Right now, if we call AssignInputValues(B, 0, num_of_classes) it does just that.

For a moment, I was confused whether it should be AssignInputValues(B, 0, num_of_classes-1)